src/packages/cli/src/args.ts

```diff
@@ -1,130 +1,168 @@
+          ] as unknown) as Definitions<Base.Config>;
+          const state = {};
+          for (const option in categoryObj) {
             const optionObj = categoryObj[option];
             if (optionObj.disableInCLI !== true) {
-              const useAliases = typeof optionObj.cliAliases !== "undefined";
-              const alias = useAliases
-                ? optionObj.cliAliases
-                : (optionObj as any).legacyName;
+              let alias = optionObj.cliAliases;
 
-              let description = optionObj.shortDescription;
+              let description = highlight(optionObj.cliDescription || "");
               if (alias) {
-                description = `${description}\n${COLORS.Bold}${
-                  COLORS.FgYellow
-                }Deprecated aliases: ${
-                  Array.isArray(alias)
-                    ? alias.filter(a => a.length > 1).join(", ")
-                    : alias
-                }${COLORS.Reset}\n`;
+                description = chalk`${description}${EOL}{dim deprecated aliases: ${alias
+                  .filter(a => a.length > 1)
+                  .map(a => `--${a}`)
+                  .join(", ")}}`;
+              } else {
+                alias = [];
               }
 
-              let defaultValue;
-              try {
-                const defaultGetter = (optionObj as any).default;
-                if (defaultGetter && defaultGetter.length > 0) {
-                  defaultValue = defaultGetter();
-                }
-              } catch (e) {}
-              if (
-                optionObj.cliType === "array" &&
-                !Array.isArray(defaultValue)
-              ) {
-                // if we pass `default: undefined`, yargs will return `[ undefined ]`
-                // this just explicitly fixes array types
-
-                if (typeof defaultValue === "undefined") {
-                  defaultValue = [];
-                } else {
-                  defaultValue = [defaultValue];
-                }
-              }
+              const generateDefaultDescription = () => {
+                // default sometimes requires a config, so we set supply one
+                return (state[option] = optionObj.default
+                  ? optionObj.default(state).toString()
+                  : undefined);
+              };
+              const defaultDescription =
+                optionObj.defaultDescription || generateDefaultDescription();
 
-              flavorArgs = flavorArgs.option(`${category}.${option}`, {
+              // we need to specify the type of each array so yargs properly casts
+              // the types held within each array
+              const { cliType } = optionObj;
+              const array = cliType && cliType.startsWith("array:"); // e.g. array:string or array:number
+              const type = (array
+                ? cliType.slice(6) // remove the "array:" part
+                : cliType) as YargsPrimitiveCliTypeStrings;
```

Comment:
does yargs handle `null` appropriately for `type` (aka treats it as it wasn't provided)? i'd probably do something like `: (cliType || undefined)` to force `type` to be `undefined` if it was `null`

Reply:
It does handle it. It _only_ checks if type is a valid types (boolean, string, number, etc), `null,` `undefined`, "this is a shoe", etc, would be ignored.